Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt

```diff
@@ -0,0 +1,138 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.os.Parcelable
+import androidx.annotation.LayoutRes
+import androidx.annotation.StringRes
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.RawValue
+
+@DslMarker
+annotation class CwaDialogDsl
+
+typealias DialogAction = () -> Unit
+
+@Parcelize
+data class CwaDialogTexts(
+    val title: @RawValue IntOrString? = null,
+    val message: @RawValue IntOrString? = null,
+    val positiveButtonText: @RawValue IntOrString? = null,
+    val negativeButtonText: @RawValue IntOrString? = null,
+    val neutralButtonText: @RawValue IntOrString? = null,
+) : Parcelable
+
+@CwaDialogDsl
+class CwaDialogBuilder {
```

Comment:
maybe in another PR , could you write some unit tests for this class ?

Reply:
Alright.